File: python_backend/app/application/use_cases/get_subscription.py
"""GetSubscriptionUseCase — current subscription for a user (or None)."""
from __future__ import annotations

from typing import Optional

from app.application.ports.subscriptions_repository import SubscriptionsRepository
from app.domain.entities.subscription import Subscription


class GetSubscriptionUseCase:
    def __init__(self, subscriptions_repo: SubscriptionsRepository) -> None:
        self.subscriptions = subscriptions_repo

    def execute(self, *, user_id: str) -> Optional[Subscription]:
        return self.subscriptions.get_active_for_user(user_id)

File: apps/web/src/application/use-cases/get-subscription.ts
/**
 * GetSubscriptionUseCase — fetch the current user's active subscription.
 *
 * Returns null when the user has no subscription yet (auth middleware
 * auto-starts the 14-day trial on first authenticated request, so the
 * common case post-F1 is `trialing`).
 */

import type { BillingApiPort } from "@/application/ports/billing-api.port";
import type { Subscription } from "@/domain/entities/subscription";

export class GetSubscriptionUseCase {
  constructor(private readonly api: BillingApiPort) {}

  execute(): Promise<Subscription | null> {
    return this.api.getSubscription();
  }
}
